File: hpcpycode/auc_serial.py
#!/usr/bin/env python

# auc_serial.py

import sys

def main():
 if len(sys.argv) < 2:
    print ("Error: must supply a number of points as a command line argument")
 else:
    ntot  = int(sys.argv[1])
    width = 3.0
    dx    = width/ntot
    x = 0
    a = 0.0
    for i in range(ntot):
        y = 0.7*x**3 - 2*x**2 + 4
        a += y*dx
        x += dx
    print("The area is %f"%a)

if __name__ == "__main__":
  main()

File: hpcpycode/auc_serial.cpp
// auc_serial.cpp

#include <iostream>
#include <cmath>
#include <cstdlib>

int main(int argc, char** argv)
{
    const size_t ntot = (argc>1)?atoi(argv[1]):100;
    const double width = 3.0;
    const double dx = width/ntot;

    double x = 0;
    double a = 0.0;

    for (size_t i=0; i<ntot; ++i) {
        const double y = 0.7*x*x*x - 2*x*x + 4;
        a += y*dx;
        x += dx;
    }

    std::cout << "The area is " << a << std::endl;

    return 0;
}

